File: spade8/1st_week/1931_conference_room.py
conference_num = int(input())
conference =[]

for i in range(conference_num):
    conference.append(list(map(int, input().split())))
##데이터 입력

conference.sort(key=lambda  x: (x[1],x[0]))
##데이터 정렬

num= 1
end = conference[0][1]
for j in range(1, conference_num):
    if conference[j][0] >= end:
        num = num + 1
        end = conference[j][1]


print(num)
##결과 출력

File: zhilly11/BOJ/Greedy/1931_Conference-room.swift
//
//  [백준] 1931.swift
//  CodingStudy
//
//  Created by 최지혁 on 2022/05/10.
//

import Foundation

let n = Int(readLine()!)!
var talks = [(Int,Int)]()

for _ in 0..<n {
    let input = readLine()!.split(separator: " ").map{Int(String($0))!}
    let (s,e) = (input[0],input[1])
    talks.append((s,e))
}

talks.sort{
    if $0.1 == $1.1 {
        return $0.0 < $1.0
    }
    return $0.1 < $1.1
}

var end = 0
var cnt = 0

for (s,e) in talks {
    if s >= end {
        end = e
        cnt += 1
    }
}
print(cnt)
